src/network_engine.cpp

```diff
@@ -814,12 +814,14 @@ NetworkEngine::sendListen(std::shared_ptr<Node> n, const InfoHash& infohash, con
     msgpack::packer<msgpack::sbuffer> pk(&buffer);
     pk.pack_map(5+(network?1:0));
 
-    pk.pack(std::string("a")); pk.pack_map(3 +
-                               (query.where.getFilter() or not query.select.getSelection().empty() ? 1:0));
+    auto has_query = query.where.getFilter() or not query.select.getSelection().empty();
+    pk.pack(std::string("a")); pk.pack_map(3 + (has_query ? 1:0));
```

Comment:
(has_query ? 1:0)  -> + has_query 
Since has_query is a boolean is will be convert into 1 or 0 (according to true / false)